app/models/user.rb

```diff
@@ -72,15 +72,15 @@ def completed_codewar_for_lesson(lesson)
   end
 
   def as_json(options = {})
-    options ||= { only: [:id,
-                         :created_at,
-                         :teacher,
-                         :school_id,
-                         :admin,
-                         :codewars_username,
-                         :homepage,
-                         :github_username] }
-    super options
+    default_options = { only: [:id,
+                               :created_at,
+                               :teacher,
+                               :school_id,
+                               :admin,
+                               :codewars_username,
+                               :homepage,
+                               :github_username] }
+    super(default_options.merge! options)
```

Comment:
Two things: instead of `default_options.merge`, call `options.merge`, otherwise you are overwriting the passed in hash values. Also, there's no need to use `merge!` since you are using the return value of `default_options.merge`.